cpp/src/arrow/builder.cc

```diff
@@ -875,7 +897,7 @@ Status BooleanBuilder::AppendValues(const std::vector<bool>& values) {
   internal::GenerateBitsUnrolled(raw_data_, length_, length,
                                  [values, &i]() -> bool { return values[i++]; });
 
-  // this updates length_
+  // this updates length_ and sets unfinished
```

Comment:
Is setting unfinished here needed?

Reply:
Since unfinished signals that there were values appended after the last `FinishInternal` call, I would say yes.